### PR TITLE
ch-(tailwind)-Integrate tailwind-cp8

### PR DESCRIPTION
**Pull request by feyton**
## What this PR do:

- [x] Successfully add tailwind CSS to the project
- [x] Replace all prior styles with the tailwind classes
- [x] Successfully add primary colors  to the theme

## Description of Task to be completed
Set up Tailwind CSS in the React app

- [x] Tailwind CSS is setup in the project
- [x] Sample styles are rendering using Tailwind CSS

### How should this be manually tested?
Check the prior requirements on our [page here](https://github.com/atlp-rwanda/phantom-fe-codebandits#readme)
```
// build UI components and use the tailwind CSS
// Linked below are the resources
npm i
npm start
// check your renders for applied styles
```
**[Tailwind states](https://tailwindcss.com/docs/hover-focus-and-other-states)**
## Recommendations

For better development workflow, I recommend:
- Installing [Tailwind Extension for VS Code from here](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss)
- Change VS Code settings to allow `@ rules for CSS` by adding this entry into `settings.json` < `"css.lint.unknownAtRules": "ignore",`.> **To access settings**: press `CTRL + P` then type `settings.json`

## Any background context you want to provide?
The integration was successfully with the use of **Postcss** as the web-pack loader
According to the docs, Tailwind require no **CSS preprocessor** but given the nature of the project the use was required.

As an addition:
**The following files has to be configure as `common js` files to avoid web-pack errors. It does not affect their role.** 
- **tailwing.config.js** to **`tailwind.config.cjs`**
- **postcss.config.js** to **`postcss.config.cjs`**

> Trying to convert these files into ES6 modules broke the functionality

## What are the relevant stories
- **[#cp8](https://trello.com/c/LKBFgNmn)**

### CODE REVIEW

**Review by samyshyaka — COMMENTED** [review]
Great job Fab,

Should the colors be limited to only those specified in the tailwind.config.cjs? The others don't seem to work

**Comment by feyton** [comment]
> Great job Fab,
> 
> Should the colors be limited to only those specified in the tailwind.config.cjs? The others don't seem to work

@samuel-shyaka-dus  When the colors are defined, you are limited to your themed/defined colors. To access all colors and their shade like `bg-blue-500` you remove `color` entry into the theme. Since we have like only 4 colors and 4 shades, it would be great to define the colors that we will be using. Adding a `key:value` pair in the colors make the color available directly.

File: src/app.css [hunk 1]
```diff
@@ -1,29 +1,4 @@
...
-  margin: 0 8px;
-}
-
-.reset {
-  margin-left: 15px;
-}
+@import url('https://fonts.googleapis.com/css2?family=Open+Sans:wght@300&family=Raleway:wght@400;700&display=swap');
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
```
Comment (Yvad60) [inline src/app.css:4]: what do these lines of code in app.css do?

**Review by Yvad60 — COMMENTED** [review]
Awesome! styles are working fine, predefined properties in `tailwind.config.cjs` are customizable and working as expected

File: src/app.css (hunk 1 above)
Comment (feyton) [reply src/app.css:4]: We are importing the tail wind css utilities to be included in our project through our css

File: tailwind.config.cjs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+module.exports = {
+  content: ['./src/**/*.{js, jsx}'],
```
Comment (Jkadhuwa) [inline tailwind.config.cjs:2]: You can add a purge:[ ] option so as to discard unused css

**Review by Jkadhuwa — CHANGES_REQUESTED** [review]
Nice work, you saved us a lot of man-hours,  look into the feedback I left below and the PR will be ready for merging.

**Comment by Jkadhuwa** [comment]
> > Great job Fab,
> > Should the colors be limited to only those specified in the tailwind.config.cjs? The others don't seem to work
> 
> @samuel-shyaka-dus When the colors are defined, you are limited to your themed/defined colors. To access all colors and their shade like `bg-blue-500` you remove `color` entry into the theme. Since we have like only 4 colors and 4 shades, it would be great to define the colors that we will be using. Adding a `key:value` pair in the colors make the color available directly.

@feyton You can extend the existing colors by using extend in the theme object but if the team is interested in just the 4 colors then the setup is okay.

File: tailwind.config.cjs (hunk 2 above)
Comment (feyton) [reply tailwind.config.cjs:2]: @Jkadhuwa 
Tailwind automatically purge CSS in production. Added two new scripts to test the difference and ensured that build is run with mode set to production.
[Reference](https://v2.tailwindcss.com/docs/optimizing-for-production#:~:text=now%20whenever%20you%20compile%20your%20css%20with)

Also the **`purge`** option has been changed to content now. And purge is performed based on the production mode.

I run quite a few tests for a build with production mode and another with dev mode. Below is the difference in file size being served:

#### Production mode:
image

#### Dev mode:
image


> And I changed the configuration to extend on the tailwind configuration for both Font family and colors

**Review by Jkadhuwa — DISMISSED** [review]

**Review by Jkadhuwa — APPROVED** [review]

**Pull request merged**